cluster-autoscaler/cloudprovider/azure/azure_container_service_pool.go

```diff
@@ -307,24 +314,55 @@ func (agentPool *ContainerServiceAgentPool) MinSize() int {
 //TargetSize gathers the target node count set for the cluster by
 //querying the underlying service.
 func (agentPool *ContainerServiceAgentPool) TargetSize() (int, error) {
-	return agentPool.GetNodeCount()
+	agentPool.mutex.Lock()
+	defer agentPool.mutex.Unlock()
+
+	if agentPool.lastRefresh.Add(15 * time.Second).After(time.Now()) {
+		return agentPool.curSize, nil
```

Comment:
is it possible that the returned target size is not accurate since it's cached? what behavior would happen if it's a wrong num?

Reply:
yep, if user invokes the scaling operation on other places, then the count may be wrong. In that case, autoscaler would correct it back based on resource usage.